src/parser.rs

```diff
@@ -181,107 +186,132 @@ impl Decoder for ValueCodec {
     }
 }
 
-pub struct ValueFuture<R> {
-    reader: Option<R>,
-    state: AnySendPartialState,
-    // Intermediate storage for data we know that we need to parse a value but we haven't been able
-    // to parse completely yet
-    remaining: Vec<u8>,
+// https://github.com/tokio-rs/tokio/pull/1687
+async fn fill_buf<R>(reader: &mut R) -> io::Result<&[u8]>
```

Comment:
it can simplify like this, the `transmute` is only to Extend the life time
```rust
async fn fill_buf<R>(reader: &mut R) -> io::Result<&[u8]>
where
    R: AsyncBufRead + Unpin,
{
    future::poll_fn(|cx| unsafe { std::mem::transmute(Pin::new(&mut *reader).poll_fill_buf(cx)) })
        .await
}
```